### PR TITLE
Containerize Scooze

### LINKED ISSUE #318: Dockerize Scooze
### Background

Originally discussed in #291. We wanted to host Scooze and set it up as a saas product to let people use it that way, but we've shifted our goal here to setting up Scooze as a locally-running Docker container that anyone can spin up and use as needed.

### Acceptance Criteria

1. Create a Dockerfile for Scooze
2. Create a docker-compose file that sets up and runs Scooze and Mongo
3. Update the `scooze setup docker` command to run docker compose

### PR DESCRIPTION
**Pull request by cmentzer**
Adds docker-compose support for running containerized versions of the Scooze REST API and MongoDB dependency

### CODE REVIEW

**Comment by iambroadband** [comment]
blocked by #294

File: src/scooze/config.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 # region Flags
 
 DEBUG = os.getenv("SCOOZE_DEBUG", "False") == "True"
+MONGO_HOST = os.getenv("MONGO_HOST", "127.0.0.1")
```
Comment (fjork3) [inline src/scooze/config.py:11]: Do we want to do this for the port as well, or is 27017 globally standard enough to leave as constant?

File: pyproject.toml [hunk 2]
```diff
@@ -38,6 +38,7 @@ pydantic = ">=2.0.0"
 pydantic-settings = "^2.0.0"
 python = ">=3.11, <4"
 requests = "^2.31.0"
+setuptools = "^68.2.2"
```
Comment (fjork3) [inline pyproject.toml:40]: Do we need this as a general requirement, or just for dev? (I see this is moving it over, curious what relies on it for user-facing stuff.)

Comment (iambroadband) [reply pyproject.toml:40]: `import pkg_resources` is used in the CLI to get the installed version of the package.

File: src/scooze/config.py (hunk 1 above)
Comment (iambroadband) [reply src/scooze/config.py:11]: https://www.mongodb.com/docs/manual/reference/default-mongodb-port/

File: src/scooze/config.py [hunk 3]
```diff
@@ -24,7 +25,7 @@ class Version(NamedTuple):
 
 class ScoozeSettings(BaseSettings):
     _version: Version = Version(*tuple(importlib.metadata.version("scooze").split(".")))
-    mongo_dsn: str = "mongodb://127.0.0.1:27017"
+    mongo_dsn: str = f"mongodb://{MONGO_HOST}:27017"
```
Comment (iambroadband) [inline src/scooze/config.py:28]: 😂

**Review by JacobGinsparg — DISMISSED** [review]

**Comment by iambroadband** [comment]
Follow-up work: https://github.com/arcavios/scooze/issues/317 and https://github.com/arcavios/scooze/issues/320

File: README.md [hunk 4]
```diff
@@ -33,9 +32,7 @@ Install using `pip install scooze`. For more installation options, see the [Inst
 ## A Simple Example
 
 ``` python
-from scooze.card import Card
-from scooze.deck import Deck, InThe
-from scooze.catalogs import Format
+from scooze.card import Card, Deck, Format, InThe
```
Comment (fjork3) [inline README.md]: just `from scooze`

**Review by fjork3 — APPROVED** [review]

**Pull request merged**